app/services/full_refresh_upload_task_service.rb

```diff
@@ -37,6 +37,14 @@ def task_options
     end
   end
 
+  def task_input
+    if @source.previous_sha.present? && @source.previous_size.present?
```

Comment:
Do you really need to explicitly check the presence here? The constructor sets them to nil. The end result is the same.

Reply:
@hsong-rh @bzwei In Go there is an omit empty which removes the attributes that have null values from the payload. I think Hui is trying to achieve that here but the default value for the 2 arguments (sha and size) are nil so they would still be converted to JSON and sent out. The whole CatalogInventory::Payload object is being converted to json, should it a hash in the object

Reply:
@bzwei you are right, no need to check. Keep logic only in parent class